testsuite/testsuite.c

```diff
@@ -98,7 +98,16 @@ int test_init(const struct test *start, const struct test *stop, int argc,
 			help();
 			return 0;
 		case 'n':
+			if (oneshot) {
+				ERR("oneshot can be provided only once\n");
+				return -1;
+			}
 			oneshot = 1;
+			/*
+			 * The option requires an argument, thus getopt will return an
+			 * index just past it... Even though we want the argument's index.
+			 */
+			optind--;
```

Comment:
not a real problem in this case since the other options stop parsing... but modifying optind like this is error prone if you can do things like `foo --foo -n test --bar`.

probably better to leave the optind alone an check for number of arguments in the different  ways depending on the oneshot option, i.e. the test name is still a a positional arg.
```